modules/tensor_mechanics/src/materials/MultiParameterPlasticityStressUpdate.C

```diff
@@ -0,0 +1,949 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "MultiParameterPlasticityStressUpdate.h"
+
+#include "Conversion.h"      // for stringify
+#include "libmesh/utility.h" // for Utility::pow
+
+template <>
+InputParameters
+validParams<MultiParameterPlasticityStressUpdate>()
+{
+  InputParameters params = validParams<StressUpdateBase>();
+  params.addClassDescription("Return-map and Jacobian algorithms for plastic models where the "
+                             "yield function and flow directions depend on multiple functions of "
+                             "stress");
+  params.addParam<std::string>("name_prepender",
```

Comment:
We call this parameter `base_name` in TensorMechanics, just FYI (make of it what you will)